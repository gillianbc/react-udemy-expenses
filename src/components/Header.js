// Stateless functional component
import React from "react";
import {NavLink} from "react-router-dom";

const Header = () => (
  <header>
    <h1>Expensify App</h1>
    <p><NavLink to="/" exact={true} activeClassName="isactive">DASHBOARD</NavLink></p>
    <p><NavLink to="/create" activeClassName="isactive">CREATE</NavLink></p>
    <p><NavLink to="/help" activeClassName="isactive">HELP</NavLink></p>
  </header>
)

export default Header