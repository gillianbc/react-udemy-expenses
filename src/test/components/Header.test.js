import React from 'react'
import Header from '../../components/Header'
import { shallow } from 'enzyme'
/*  Using ReactShallowRenderer and a snapshot
import ReactShallowRenderer from 'react-test-renderer/shallow'
it('rendered header should match the snapshot', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Header />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
})*/

it('There should be 1 h1 element', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('h1').length).toBe(1)
})

it('The h1 text should be correct', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('h1').text()).toBe('Expensify App')
})

it('The snapshot should match', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
})



