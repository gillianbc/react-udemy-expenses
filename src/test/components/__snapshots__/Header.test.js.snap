// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The snapshot should match 1`] = `
<header>
  <h1>
    Expensify App
  </h1>
  <p>
    <NavLink
      activeClassName="isactive"
      exact={true}
      to="/"
    >
      DASHBOARD
    </NavLink>
  </p>
  <p>
    <NavLink
      activeClassName="isactive"
      to="/create"
    >
      CREATE
    </NavLink>
  </p>
  <p>
    <NavLink
      activeClassName="isactive"
      to="/help"
    >
      HELP
    </NavLink>
  </p>
</header>
`;
